File: src/main/java/thedarkcolour/futuremc/block/BlockLoom.java
package thedarkcolour.futuremc.block;

import net.minecraft.block.SoundType;
import net.minecraft.block.material.Material;
import net.minecraft.block.state.IBlockState;
import net.minecraft.creativetab.CreativeTabs;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.util.EnumFacing;
import net.minecraft.util.EnumHand;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;
import thedarkcolour.core.block.BlockRotatable;
import thedarkcolour.core.gui.Gui;
import thedarkcolour.futuremc.FutureMC;
import thedarkcolour.futuremc.init.FutureConfig;

public class BlockLoom extends BlockRotatable {
    public BlockLoom() {
        super("Loom", Material.WOOD, SoundType.WOOD);
        setCreativeTab(FutureConfig.general.useVanillaTabs ? CreativeTabs.DECORATIONS : FutureMC.TAB);
    }

    @Override
    public boolean onBlockActivated(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ) {
        if (FutureMC.DEBUG && !worldIn.isRemote) {
            Gui.LOOM.open(playerIn, worldIn, pos);
            return true;
        }
        return false;
    }
}

File: src/main/java/thedarkcolour/futuremc/block/BlockLoom.kt
package thedarkcolour.futuremc.block

import net.minecraft.block.state.IBlockState
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.util.EnumFacing
import net.minecraft.util.EnumHand
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World
import thedarkcolour.core.block.RotatableBlock
import thedarkcolour.core.gui.Gui

class BlockLoom(properties: Properties?) : RotatableBlock(properties) {
    override fun onBlockActivated(
        worldIn: World, pos: BlockPos, state: IBlockState, playerIn: EntityPlayer,
        hand: EnumHand, facing: EnumFacing, hitX: Float, hitY: Float, hitZ: Float
    ): Boolean {
        if (!worldIn.isRemote) {
            Gui.LOOM.open(playerIn, worldIn, pos)
            return true
        }
        return true
    }
}